pkg/driver/aws-efs/aws_efs_tags_controller.go

```diff
@@ -0,0 +1,391 @@
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/aws/aws-sdk-go/service/sts"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorapi "github.com/openshift/api/operator/v1"
+	"github.com/openshift/csi-operator/pkg/clients"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+const (
+	awsEFSSecretNamespace = "openshift-cluster-csi-drivers"
+	awsEFSSecretName      = "aws-efs-cloud-credentials"
+	efsDriverName         = "efs.csi.aws.com"
+	tagHashAnnotationKey  = "efs.openshift.io/access-point-tags-hash"
+	infrastructureName    = "cluster"
+
+	defaultReSyncPeriod = 30 * time.Minute
+)
+
+type EFSAccessPointTagsController struct {
+	name          string
+	commonClient  *clients.Clients
+	eventRecorder events.Recorder
+	queue         workqueue.TypedRateLimitingInterface[string]
+	queueSet      map[string]struct{} // A set to track added volume names
+	mutex         sync.Mutex
+}
+
+// TokenClaims represents the JWT claims
+type TokenClaims struct {
+	Exp int64 `json:"exp"` // Expiry timestamp
+}
+
+func NewEFSAccessPointTagsController(
+	name string,
+	commonClient *clients.Clients,
+	eventRecorder events.Recorder) factory.Controller {
+
+	c := &EFSAccessPointTagsController{
+		name:          name,
+		commonClient:  commonClient,
+		eventRecorder: eventRecorder,
+		queue:         workqueue.NewTypedRateLimitingQueue[string](workqueue.NewTypedItemExponentialFailureRateLimiter[string](10*time.Second, 100*time.Hour)),
+		queueSet:      make(map[string]struct{}),
+		mutex:         sync.Mutex{},
+	}
+	return factory.New().WithSync(
+		c.Sync,
+	).WithInformers(
+		c.commonClient.ConfigInformers.Config().V1().Infrastructures().Informer(),
+	).ResyncEvery(
+		defaultReSyncPeriod,
+	).WithPostStartHooks(
+		c.startTagsQueueWorker,
+	).ToController(
+		name,
+		eventRecorder,
+	)
+}
+
+func (c *EFSAccessPointTagsController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	klog.Infof("EFSAccessPointTagsController sync started")
+	defer klog.Infof("EFSAccessPointTagsController sync finished")
+
+	opSpec, _, _, err := c.commonClient.OperatorClient.GetOperatorState()
+	if err != nil {
+		return err
+	}
+	if opSpec.ManagementState != operatorapi.Managed {
+		return nil
+	}
+
+	infra, err := c.getInfrastructure()
+	if err != nil {
+		return err
+	}
+	if infra == nil || infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || !isHypershiftCluster(infra) {
+		return nil
+	}
+	err = c.processInfrastructure(ctx, infra)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// isHypershiftCluster validates the cluster is hypershift cluster or not
+func isHypershiftCluster(infra *configv1.Infrastructure) bool {
+	if infra.Status.ControlPlaneTopology == configv1.ExternalTopologyMode {
```

Comment:
AFAIK, `ExternalTopologyMode` is not a reliable indicator of a hypershift cluster because it may be used for other flavors of openshift as well. IBM ROKS [for example](https://github.com/openshift/cluster-storage-operator/blob/30a471d023eb0e58d6b9735fadb7711d3bdb5b2a/pkg/operator/csidriveroperator/csioperatorclient/ibm-vpc-block.go#L22-L24) uses it (but of course aws-ebs is not running on IBM ROKS).

I think it would be better to check the `hypershift.openshift.io/managed: "true"` label if the goal is to enable it only on hypershift deployments.